code/modules/mob/living/carbon/give.dm

```diff
@@ -1,52 +1,164 @@
+	if(HAS_TRAIT(src, TRAIT_OFFERING_ITEM))
+		to_chat(src, "<span class='warning'>You're already offering an item to someone!</span>")
+		return
+	if(istype(client.click_intercept, /datum/click_intercept/give))
+		give_icon.icon_state = "act_give_off"
+		QDEL_NULL(client.click_intercept)
+		return
+	var/obj/item/I = get_active_hand()
+	if(!I)
+		to_chat(src, "<span class='warning'>You don't have anything in your hand to give!</span>")
+		return
+	if(I.flags & NODROP)
+		to_chat(src, "<span class='warning'>[I] is stuck to your hand, you can't give it away!</span>")
+		return
+	if(I.flags & ABSTRACT)
+		to_chat(src, "<span class='warning'>That's not exactly something you can give.</span>")
+		return
+
+	give_icon.icon_state = "act_give_on"
+	new /datum/click_intercept/give(client)
+
+/**
+ * # Give click intercept
+ *
+ * While a mob has this intercept, left clicking on a carbon mob will attempt to offer their currently held item to that mob.
+ */
+/datum/click_intercept/give
 
-	if(!iscarbon(target)) //something is bypassing the give arguments, no clue what, adding a sanity check JIC
-		to_chat(usr, "<span class='danger'>Wait a second... \the [target] HAS NO HANDS! AHH!</span>")//cheesy messages ftw
+/datum/click_intercept/give/New(client/C)
+	..()
+	holder.mouse_pointer_icon = file("icons/mouse_icons/give_item.dmi")
+	to_chat(holder, "<span class='info'>You can now left click on someone to give them your held item.</span>")
+	RegisterSignal(holder.mob.get_active_hand(), list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/cancel_give)
+	RegisterSignal(holder.mob, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS), .proc/cancel_give)
+
+
+/datum/click_intercept/give/Destroy(force = FALSE, silent = FALSE)
+	holder.mouse_pointer_icon = initial(holder.mouse_pointer_icon)
+	holder.mob.give_icon.icon_state = "act_give_off"
+	if(!silent)
+		to_chat(holder.mob, "<span class='info'>You're no longer trying to give someone your held item.</span>")
+	UnregisterSignal(holder.mob.get_active_hand(), list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED))
+	UnregisterSignal(holder.mob, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS))
+	return ..()
+
+
+/datum/click_intercept/give/proc/cancel_give()
+	SIGNAL_HANDLER
+	qdel(src)
+
+
+/datum/click_intercept/give/InterceptClickOn(mob/user, params, atom/object)
+	if(user == object || !iscarbon(object))
+		return
+	var/mob/living/carbon/receiver = object
+	var/obj/item/I = user.get_active_hand()
+	if(!user.Adjacent(object))
+		to_chat(user, "<span class='warning'>You need to be closer to [receiver] to offer them [I]</span>")
+		return
+	if(!receiver.client)
+		to_chat(user, "<span class='warning'>You offer [I] to [receiver], but they don't seem to respond...</span>")
 		return
+	if((receiver.r_hand && receiver.l_hand) || HAS_TRAIT(receiver, TRAIT_HANDS_BLOCKED))
+		to_chat(user, "<span class='warning'>[receiver] doesn't have any hands free to accept [I]!</span>")
```

Comment:
Of all the checks, i think this one is the most surmountable, and should not prevent you from trying to offer.
They will get the popup, click it, realize that their hands are full, fiddle with the inventory, and then click the popup again, and everyone would be happy.
But with this check here it becomes rather awkward - you need to verbally ask them to free their hands before you can offer them an item, which ruins the flow IMO.

Reply:
Fair, I'll remove it.